### PR TITLE
Add Devise gem

### PR DESCRIPTION
**Pull request by Zama15**
The gem was added, the database were change and the old versions of the file that were replaced were deleted 

## Description

- What changes are being made?
The gem devise were added, styling the views, modified the migration of the user table and added a controller which inherit to devise and add more params in the permit register
- Why are these changes necessary?
Because the lack of confident in the own logic for registration and sessions
- How have these changes been tested?
With the host PUMA to test the application

## List of Changes

1. Modified the database
   - the data were deleted due to a serie of errors occurred when adding the devise
2. Modified the views
   - Add new style to the forms and the modal
   - with the new paths some files that use the olds routes were modified
4. Modified user model
5. Add registration controller
   - The controller inherit from the devise registration controller
   - The migration were made to add new information in addition to the defaults values [username avatar]

## Additional Information

image
image
image
image

## Checklist

- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Errors

The changes made generate a new error which is the 
`ActionController::InvalidAuthenticityToken in Users::RegistrationsController#create`
The full log look like this
```
Started POST "/users" for 127.0.0.1 at 2024-02-05 16:39:47 -0600
Processing by Users::RegistrationsController#create as TURBO_STREAM
  Parameters: {"authenticity_token"=>"[FILTERED]", "user"=>{"username"=>"aaaaaaaaaaaaaa", "email"=>"", "password"=>"[FILTERED]", "password_confirmation"=>"[FILTERED]"}, "commit"=>"Sign up"}
Can't verify CSRF token authenticity.
Completed 422 Unprocessable Entity in 1ms (ActiveRecord: 0.0ms | Allocations: 712)
```

### CODE REVIEW

**Comment by JuanequeX** [comment]
These resources can help you to fix the error with the CSRF
https://github.com/rails/rails/issues/34837
https://copyprogramming.com/howto/csrf-token-error-occurs-when-using-turbolinks-with-ruby-on-rails
https://github.com/nsarno/knock/issues/208#issuecomment-373022274

File: app/assets/stylesheets/base/_base.scss [hunk 1]
```diff
@@ -33,6 +33,6 @@ input, select, textarea {
   }
 }
 
-img, a {
+img, a, label {
   user-select: none;
 }
```
Comment (juperameza) [inline app/assets/stylesheets/base/_base.scss]: New line

**Review by juperameza — CHANGES_REQUESTED** [review]
There's a better way to add stuff to devise without generating the controllers
And don't leave commented code

**Comment by juperameza** [comment]
> There's a better way to add stuff to devise without generating the controllers
> And don't leave commented code 
https://gist.github.com/withoutwax/46a05861aa4750384df971b641170407

File: app/assets/stylesheets/components/_modal.scss [hunk 2]
```diff
@@ -61,28 +62,33 @@
 .modal {
   &-sm {
     max-height: 50%;
+    max-width: 50%;
     width: 45%;
   }
 
   &-md {
     max-height: 70%;
+    max-height: 65%;
```
Comment (JuanequeX) [inline app/assets/stylesheets/components/_modal.scss]: what is the reason to maintain two `max-height`?

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/base/_base.scss (hunk 1 above)
Comment (Zama15) [reply app/assets/stylesheets/base/_base.scss]: done

File: app/assets/stylesheets/components/_modal.scss (hunk 2 above)
Comment (Zama15) [reply app/assets/stylesheets/components/_modal.scss]: my bad, have to be width, skill issue :(

**Comment by Zama15** [comment]
> > There's a better way to add stuff to devise without generating the controllers
> > And don't leave commented code
> > https://gist.github.com/withoutwax/46a05861aa4750384df971b641170407

done by moving deleting the controller and move the methods to application_controller::base, furthermore the routes were modified to the defualt

**Comment by Zama15** [comment]
> These resources can help you to fix the error with the CSRF [rails/rails#34837](https://github.com/rails/rails/issues/34837) https://copyprogramming.com/howto/csrf-token-error-occurs-when-using-turbolinks-with-ruby-on-rails [nsarno/knock#208 (comment)](https://github.com/nsarno/knock/issues/208#issuecomment-373022274)

even with the 
```
before_action :configure_permitted_parameters, if: :devise_controller?
protect_from_forgery with: :exception`
```
in the controller, the CSRF error still appears
i tried with the 
`protect_from_forgery with: :null_session`
which show the errors that before doesnt appear but the form doesnt read some params with that
in addition the devise.rb and the application were modified to try to repair the error following the issue trend but the result was the same, so i just stay as the first solution even with the obvious errors

**Pull request merged**